features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/resources/ResourcesDataStoreManager.kt

```diff
@@ -0,0 +1,120 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.resources
+
+import android.text.format.DateUtils
+import com.datadog.android.api.feature.FeatureScope
+import com.datadog.android.api.feature.FeatureSdkCore
+import com.datadog.android.api.storage.datastore.DataStoreCallback
+import com.datadog.android.core.internal.persistence.Deserializer
+import com.datadog.android.core.persistence.Serializer
+import com.datadog.android.core.persistence.datastore.DataStoreContent
+import com.datadog.android.sessionreplay.internal.ResourcesFeature.Companion.SESSION_REPLAY_RESOURCES_FEATURE_NAME
+import com.datadog.android.sessionreplay.model.ResourceHashesEntry
+import java.util.Collections
+
+internal class ResourcesDataStoreManager(
+    private val featureSdkCore: FeatureSdkCore,
+    private val resourceHashesSerializer: Serializer<ResourceHashesEntry>,
+    private val resourcesHashesDeserializer: Deserializer<String, ResourceHashesEntry>,
+    private val featureScope: FeatureScope? = featureSdkCore.getFeature(
+        SESSION_REPLAY_RESOURCES_FEATURE_NAME
+    )
+) {
+    private val knownResources = Collections.synchronizedSet(mutableSetOf<String>())
+    private var storedLastUpdateDateNs = System.nanoTime()
+
+    init {
+        initializeManager()
+    }
+
+    internal fun wasResourcePreviouslySent(resourceHash: String): Boolean =
+        knownResources.contains(resourceHash)
+
+    internal fun store(resourceHash: String) {
+        knownResources.add(resourceHash)
+        writeResourcesToStore()
```

Comment:
I think we are missing atomicity property here. What if hash was added to the `knownResources`, but it wasn't written to the store by some reason? Does it have any impact?

Maybe we can first write to the store, and then, if we know that write operation succeeded, update `knownResources`?

Reply:
🤔  do we want to make them dependent? The datastore copy is the persistent copy of the known hashes between sessions, while the knownResources is for the current session. Even if we failed to save to persistent storage, we've still seen this hash for the current session and should avoid sending it

Reply:
Don't know, I'm just raising the question if we need atomicity here or not.